contracts/matic/SubscriptionManager.sol

```diff
@@ -0,0 +1,148 @@
+    mapping (bytes16 => Policy) internal _policies;
+
+    function initialize(uint256 _feeRate) public initializer {
+        _setFeeRate(_feeRate);
+        _setupRole(SET_RATE_ROLE, msg.sender);
+        _setupRole(WITHDRAW_ROLE, msg.sender);
+        _setupRole(DEFAULT_ADMIN_ROLE, msg.sender);
+    }
+
+    function getPolicyCost(
+        uint16 _size,
+        uint32 _startTimestamp,
+        uint32 _endTimestamp
+    ) public view returns (uint256){
+        uint32 duration = _endTimestamp - _startTimestamp;
+        require(duration > 0, "Invalid timestamps");
+        require(_size > 0, "Invalid policy size");
+        return feeRate * _size * duration;
+    }
+
+    function createPolicy(
+        bytes16 _policyId,
+        address _policyOwner,
+        uint16 _size,
+        uint32 _startTimestamp,
+        uint32 _endTimestamp
+    )
+        external payable
+    {
+        require(
+            _startTimestamp < _endTimestamp && block.timestamp < _endTimestamp,
+            "Invalid timestamps"
+        );
+        require(
+            msg.value == getPolicyCost(_size, _startTimestamp, _endTimestamp),
+            "Invalid policy cost"
+        );
+
+        _createPolicy(_policyId, _policyOwner, _size, _startTimestamp, _endTimestamp);
+    }
+
+    /**
+    * @notice Create policy
+    * @param _policyId Policy id
+    * @param _policyOwner Policy owner. Zero address means sender is owner
+    * @param _size Number of nodes involved in the policy
+    * @param _startTimestamp Start timestamp of the policy in seconds
+    * @param _endTimestamp End timestamp of the policy in seconds
+    */
+    function _createPolicy(
+        bytes16 _policyId,
+        address _policyOwner,
+        uint16 _size,
+        uint32 _startTimestamp,
+        uint32 _endTimestamp
+    )
+        internal returns (Policy storage policy)
+    {
+        policy = _policies[_policyId];
+        require(
+            policy.endTimestamp < block.timestamp,
+            "Policy is currently active"
+        );
+
+        policy.sponsor = payable(msg.sender);
+        policy.startTimestamp = _startTimestamp;
+        policy.endTimestamp = _endTimestamp;
+        policy.size = _size;
+
+        if (_policyOwner != msg.sender && _policyOwner != address(0)) {
+            policy.owner = _policyOwner;
+        }
+
+        emit PolicyCreated(
+            _policyId,
+            msg.sender,
+            _policyOwner == address(0) ? msg.sender : _policyOwner,
+            _size,
+            _startTimestamp,
+            _endTimestamp
+        );
+    }
+
+    function getPolicy(bytes16 _policyID) public view returns(Policy memory){
+        return _policies[_policyID];
+    }
+
+    function isPolicyActive(bytes16 _policyID) public view returns(bool){
+        return _policies[_policyID].endTimestamp > block.timestamp;
+    }
+
+    function _setFeeRate(uint256 newFee) internal {
+        uint256 oldFee = feeRate;
```

Comment:
Do we need a `require` here to ensure that newFee is always > 0?

Reply:
Not necessarily. A fee rate of 0 is plausible.

Reply:
👍  - Que barato!